pkg/server/router/credential.go

```diff
@@ -238,6 +238,8 @@ type UpdateCredentialStatusResponse struct {
 	Revoked bool `json:"revoked"`
 }
 
+var mutex sync.Mutex
```

Comment:
I believe this wouldn't work if there are multiple instances of `ssi-service` running with the same backing DB. 

Have you considered using DB transactions instead to update the credential status? You can see how https://github.com/andresuribe87/ssi-service/blob/8f69172d96017ba9f3bcb1ee263d1b6a412a12c7/pkg/storage/bolt.go#L207 does it for an example.

Reply:
that is true, 🤔 

in our upcoming db interface change to have the ability connect to many dbs we will have to account for this in all dbs (some dbs may not support this)

In this case of unpredictable dbs it may be better to do it at the service level with mutex's. 

What do you think?

Reply:
maybe have it in both spots to try to account for both scenarios ?

Reply:
i think going with a txns model makes sense...whatever solution we go with we'll need that capability across dbs